src/phoenix/datasets/experiments.py

```diff
@@ -321,21 +339,29 @@ def evaluate_experiment(
             .get("examples", [])
         )
     ]
-    _evaluate_experiment(experiment, evaluators, dataset_examples, client)
-
-
-ExperimentEvaluatorName: TypeAlias = str
+    _evaluate_experiment(
+        experiment,
+        evaluators=evaluators,
+        dataset_examples=dataset_examples,
+        client=client,
+    )
 
 
 def _evaluate_experiment(
     experiment: Experiment,
-    evaluators: Union[ExperimentEvaluator, Iterable[ExperimentEvaluator]],
+    *,
+    evaluators: Optional[Evaluators] = None,
+    llm_evaluators: Optional[Evaluators] = None,
     dataset_examples: Iterable[Example],
     client: httpx.Client,
 ) -> None:
-    if isinstance(evaluators, (CanEvaluate, CanAsyncEvaluate)):
-        evaluators = [evaluators]
-
+    evaluators_by_name = _evaluators_by_name(
```

Comment:
It seems like we could use lists to simplify `_evaluators_by_name`.

Reply:
We need the names to distinct, hence the `dict`.

Reply:
You're saying to check whether a name is already in the list?

Reply:
You could return a list from `_evaluators_by_name` and avoid branching logic inside the function. You'd probably still use a dictionary inside the function. Non-blocking.

Reply:
Not sure I follow. Using a dict is pretty much the first thing we would use for asserting something is unique. Using a list doesn't carry the same signals.

Reply:
I imagine `_evaluators_by_name` is going to look different if we remove `llm_evaluators` as an argument. Let's circle back.

Reply:
It's still the same, but just doing in one pass instead of two